File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/ResourceAssignments/assignmentvariance.rb
module Asposetasksjava
  module AssignmentVariance
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        resource_assignments = project.getResourceAssignments().toList()

        # Parse through all the collected resource assignments
        i = 0
        while i < resource_assignments.size()
          ra = resource_assignments.get(i)
          puts "Work Variance: " + ra.get(Rjb::import('com.aspose.tasks.Asn').WORK_VARIANCE).toString()
          puts "Cost Variance: " + ra.get(Rjb::import('com.aspose.tasks.Asn').COST_VARIANCE).toString()
          puts "Start Variance: " + ra.get(Rjb::import('com.aspose.tasks.Asn').START_VARIANCE).toString()
          puts "Finish Variance: " + ra.get(Rjb::import('com.aspose.tasks.Asn').FINISH_VARIANCE).toString()
          puts "--------------------------------------------------------"
          i +=1
        end    
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResourceAssignments/AssignmentVariance.php
<?php
namespace Aspose\Tasks\WorkingWithResourceAssignments;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Asn as Asn;
class AssignmentVariance{

    public static function run($dataDir=null)
    {

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $resource_assignments = $project->getResourceAssignments()->toList();

        # Parse through all the collected resource assignments
        $i = 0;
        $asn=new Asn();
        while ($i < sizeof($resource_assignments)){
            $ra = $resource_assignments->get($i);
          print "Work Variance: " . (string)$ra -> get($asn -> WORK_VARIANCE).PHP_EOL;// . toString()
          print "Cost Variance: " . (string)$ra -> get($asn -> COST_VARIANCE).PHP_EOL;// . toString()
          print "Start Variance: " . (string)$ra -> get($asn -> START_VARIANCE).PHP_EOL;// . toString()
          print "Finish Variance: " . (string)$ra -> get($asn -> FINISH_VARIANCE).PHP_EOL;// . toString()
          print "--------------------------------------------------------".PHP_EOL;
          $i += 1;
        }

    }
}
?>